application/mounts.py

```diff
@@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
 		mount_uri = mount.get_root().get_uri()
 		volume = mount.get_volume()
 
+		# check if mount has been added as volume already
+		mount_already_in_volume_list = False
```

Comment:
I will check it out with Debian Jessie.

Note: I will cancel my other pull request - it is getting too big and I 
think we are better off if I prepare individual self-contained feature 
'pull requests' e.g. for the text viewer fix. This should it make easier 
for you too :-)

Christian.

On 15/05/15 21:56, Mladen Mijatov wrote:
>
> In application/mounts.py 
> <https://github.com/MeanEYE/Sunflower/pull/112#discussion_r30445418>:
>
> > @@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
> >  		mount_uri = mount.get_root().get_uri()
> >  		volume = mount.get_volume()
> >
> > +		# check if mount has been added as volume already
> > +		mount_already_in_volume_list = False
>
> I think this was a bug in GIO as 14.04 is quite old. I remember fixing 
> the same issue with duplicate mounts as I thought GIO was suppose to 
> work this way. After they fixed this issue in GIO they created a 
> problem for us. If possible, please test this on some newer version 
> just to make sure we are not creating another problem.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/MeanEYE/Sunflower/pull/112/files#r30445418>.
>